plugins/weather.py

```diff
@@ -19,34 +14,50 @@ class APIError(Exception):
     PrimaryKeyConstraint('nick')
 )
 
+location_cache = {}
+
 # Define some constants
 google_base = 'https://maps.googleapis.com/maps/api/'
 geocode_api = google_base + 'geocode/json'
-
-wunder_api = "http://api.wunderground.com/api/{}/forecast/geolookup/conditions/q/{}.json"
+wunder_api = "http://api.wunderground.com/api/{}/forecast/lang:{}/geolookup/conditions/q/{}.json"
 
 # Change this to a ccTLD code (eg. uk, nz) to make results more targeted towards that specific country.
 # <https://developers.google.com/maps/documentation/geocoding/#RegionCodes>
 bias = None
 
 
-def check_status(status):
-    """
-    A little helper function that checks an API error code and returns a nice message.
-    Returns None if no errors found
-    """
-    if status == 'REQUEST_DENIED':
-        return 'The geocode API is off in the Google Developers Console.'
-    elif status == 'ZERO_RESULTS':
-        return 'No results found.'
-    elif status == 'OVER_QUERY_LIMIT':
-        return 'The geocode API quota has run out.'
-    elif status == 'UNKNOWN_ERROR':
-        return 'Unknown Error.'
-    elif status == 'INVALID_REQUEST':
-        return 'Invalid Request.'
-    elif status == 'OK':
-        return None
+class GeocodeAPIError(Exception):
+    """Raised when the geocode api returns an error message.
+    This helps error messages optionally be returned *en francais*."""
+    def __init__(self, status):
+        super()
+        self._status = status
+
+    def __str__(self):
+        if self._status == 'REQUEST_DENIED':
+            return 'The geocode API is off in the Google Developers Console.'
+        if self._status == 'ZERO_RESULTS':
+            return 'No results found.'
+        if self._status == 'OVER_QUERY_LIMIT':
+            return 'The geocode API quota has run out.'
+        if self._status == 'UNKNOWN_ERROR':
+            return 'Unknown Error.'
+        if self._status == 'INVALID_REQUEST':
+            return 'Invalid Request.'
+        return repr(self._status)
```

Comment:
Please don't override `__str__` on an exception subclass to set the text, use `super().__init__("exception text")` to do it. This is consistent with the rest of the exception API.